src/MentionsInput.js

```diff
@@ -541,15 +541,42 @@ const MentionsInput = React.createClass({
     // neglect async results from previous queries
     if(queryId !== this._queryId) return;
 
+    var type = mentionDescriptor.props.type;
     var update = {};
-    update[mentionDescriptor.props.type] = {
-      query: query,
-      mentionDescriptor: mentionDescriptor,
-      querySequenceStart: querySequenceStart,
-      querySequenceEnd: querySequenceEnd,
-      results: suggestions,
-      plainTextValue: plainTextValue
-    };
+
+    // If mention doesn't specify type try suggestion type
+    if(!type) {
+      var groups = {}
```

Comment:
Upon re-reading, I have a better understanding of what you're talking about. This method may work if unchanged. I will see if it still includes the correct `type` in the `mentions` array on `onChange` without changes to this particular function.